### PR TITLE
UnionDemandManager

### PR DESCRIPTION
**Pull request by approxit**
What I've done:
* Added demand manager that can take multiple other `get_initial_proposals` functions and merge them into "single stream"

Notable remarks:
* I'm open to suggestions for the manager's name! I'm not a fan of the name that I've proposed... 🤔

### CODE REVIEW

**Review by shadeofblue — COMMENTED** [review]
might I propose an very slightly different and yet simpler alternative? :)

https://github.com/golemfactory/golem-core-python/pull/138

File: golem/managers/demand/union.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self._get_initial_proposal_funcs = get_initial_proposal_funcs
+
+        self._lock = asyncio.Lock()
+
+        self._task_map: MutableMapping[Callable[[], Awaitable[Proposal]], asyncio.Task] = {}
+
+    @trace_span("Stopping UnionDemandManager", log_level=logging.INFO)
+    async def stop(self) -> None:
+        await ensure_cancelled_many(self._task_map.values())
+        self._task_map.clear()
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply golem/managers/demand/union.py:32]: okay, nevermind, ... that _is_ the `Manager`'s interface

File: golem/managers/demand/union.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    ) -> None:
+        self._golem = golem
+        self._get_initial_proposal_funcs = get_initial_proposal_funcs
+
+        self._lock = asyncio.Lock()
+
+        self._task_map: MutableMapping[Callable[[], Awaitable[Proposal]], asyncio.Task] = {}
+
+    @trace_span("Stopping UnionDemandManager", log_level=logging.INFO)
+    async def stop(self) -> None:
```
[the remark this answers is not in the document]
Comment (approxit) [reply golem/managers/demand/union.py:30]: Well, `golem.managers.base.Manager` in fact has a `start`/`stop` interface. :)

File: golem/managers/demand/union.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+from typing import Awaitable, Callable, MutableMapping, Optional, Sequence
+
+from golem.managers import DemandManager
+from golem.node import GolemNode
+from golem.resources import Proposal
+from golem.utils.asyncio import create_task_with_logging, ensure_cancelled_many
+from golem.utils.logging import trace_span
+
+
+class UnionDemandManager(DemandManager):
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply golem/managers/demand/union.py:12]: we've settled for `AggregatingDemandManager` instead :)

[the remark this answers is not in the document]
Comment (approxit) [reply golem/managers/demand/union.py:12]: We decided offline that we'll choose `AgregatingDemandManager`.

**Comment by approxit** [comment]
I've made a few tweaks and added an extra test, to cover documented guarantees about the order of completion (after using the queue) of saved proposals.

**Review by lucekdudek — APPROVED** [review]

**Pull request merged**